### PR TITLE
Document YAML options in docs site

### PR DESCRIPTION
**Pull request by ghost**
Signed-off-by: Eric Stoekl <ems5311@gmail.com>

This change documents options for YAML file usage in the `faas-cli`. Most of the sections are taken from `md` documents in the `faas/` repo. The "Limits and Requests" section is new.

Related issue: https://github.com/openfaas/docs/issues/12

### CODE REVIEW

**Review by alexellis — APPROVED** [review]
Instead of YAML, how about "Stack file"? LGTM otherwise

File: docs/reference/yaml.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+* environment_file - defined in zero to many external files
+
+```yaml
+  environment_file:
+    - file1.yml
+    - file2.yml
+```
+
+If you specify a variable such as "access_key" in more than one `environment_file` file then the last file in the list will take priority.
```
Comment (stefanprodan) [inline docs/reference/yaml.md]: I would not use `access_key` or `secret_key` to exemplify  environmental variables, it sends the wrong message, those should be secrets. cc @alexellis

File: docs/reference/yaml.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
+# YAML format reference
```
Comment (stefanprodan) [inline docs/reference/yaml.md]: Secrets are missing

**Review by stefanprodan — CHANGES_REQUESTED** [review]

File: docs/reference/yaml.md (hunk 1 above)
Comment (alexellis) [reply docs/reference/yaml.md]: Probably was my example? Not sure, but makes sense to update - @ericstoekl - how about something else like a non-confidential configuration item?

File: docs/reference/yaml.md (hunk 2 above)
Comment (alexellis) [reply docs/reference/yaml.md]: Secrets may be covered elsewhere, so let's mention them here and the format then link to that other page too.

**Comment by alexellis** [comment]
Good suggestions from @stefanprodan, @ericstoekl can you update and we'll merge?

**Comment by alexellis** [comment]
This shouldn't take 6 days for a simple change. I'll merge and make the changes myself.

**Comment by alexellis** [comment]
I just saw we put in our official docs this line

>  - Reserve CPU -- unsure but may be `--cpu-shares`


^ why do we have the words "unsure but may be" ??

**Comment by alexellis** [comment]
>  - TODO Request Memory

^ This was not ready for merge.

**Pull request merged**